src/tests/multihost/sssd/testlib/common/custom_log.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+import os
+from collections import defaultdict
+
+import pytest
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_configure(config: pytest.Config) -> None:
+    p = CustomLogPlugin(config)
+    config.pluginmanager.register(p, 'custom_log_plugin')
+
+
+class CustomLogPlugin:
+    """
+    Custom log class to remove log from console
+    and place it inside per-test log files
+    """
+    def __init__(self, config: pytest.Config) -> None:
+        self.config: pytest.Config = config
+        self.log_per_test: str = config.getoption("log_per_test")
+        self.tests = defaultdict(dict)
+        if self.log_per_test not in ["never", "duplicate"]:
+            self.config.option.showlocals = True
+            self.config.option.reportchars = 'a'
+            self.config.option.tbstyle = 'line'
+            self.config.option.showcapture = 'no'
+            self.config.option.capture = 'fd'
```

Comment:
Do we only want these options if we're logging to the pre-test log files?   Could some of these options be helpful even for the console output?

Reply:
These affect mostly the console logging , I am overriding/overwriting values that are hardcoded in run-pytests.py.  I am reducing what is put in console as I am manually storing these in file.
If  used outside of per-test logging, it would make harder to debug issues.